workflow/controller/operator.go

```diff
@@ -651,8 +743,30 @@ func (woc *wfOperationCtx) executeTemplate(templateName string, args wfv1.Argume
 			// scheduled (or had a create pod error). Nothing to more to do with this node.
 			return nil
 		}
+
+		// If the user has specified retries, a special "retries" non-leaf node
+		// is created. This node acts as the parent of all retries that will be
+		// done for the container. The status of this node should be "Success"
+		// if any of the retries succeed. Otherwise, it is "Failed".
+
+		// TODO(shri): Mark the current node as a "retry" node
+		// Create a new child node as the first attempt node and
+		// run the template in that node.
+		nodeToExecute := nodeName
+		if tmpl.BackoffLimit != nil && *tmpl.BackoffLimit > 0 {
+			node := woc.markNodePhase(nodeName, wfv1.NodeRunning)
+			node.MaxRetries = *tmpl.BackoffLimit
+			woc.wf.Status.Nodes[nodeID] = *node
+
+			// Create new node as child of 'node'
+			newContainerName := fmt.Sprintf("%s-%d", nodeName, time.Now().Unix())
+			woc.markNodePhase(newContainerName, wfv1.NodeRunning)
```

Comment:
executeContainer() should be the one who gets to mark the node as running (which it already does). I think marking it running now is premature.

Reply:
Done.